### PR TITLE
[Strings2] Improve string repr of Sum

### PR DESCRIPTION
**Pull request by Qottmann**
`Sum` instances are now displayed multiline and without unnecessary brackets. Further, all outputs should be copy-and-paste-able. Some examples how this now looks:

```python3
>>> qml.operation.enable_new_opmath()
>>> X(0) + X(1) + X(2) # short outputs <60 characters in single line
X(0) + X(1) + X(2)
>>> 0.5 * (X(0) @ X(1)) + 0.7 * (X(1) @ X(2)) + 0.8 * (X(2) @ X(3)) # larger outputs break into multiple lines. copy-paste-able with surrounding brackets
(
    0.5 * (X(0) @ X(1))
  + 0.7 * (X(1) @ X(2))
  + 0.8 * (X(2) @ X(3))
)
```

There are also some peculiarities that cannot be improved at the moment because they are due to the nesting and lazy execution of products and sums:
```python3
>>> 0.5 * (X(0) @ (0.5 * X(1))) + 0.7 * X(1)
0.5 * (X(0) @ (0.5 * X(1))) + 0.7 * X(1)
```

ToDo

- [x] Basic functionality
- [x] Tests
- [x] multiline threshold solution? (Problem from nesting in e.g. X(0) + X(1) + X(2))
- [x] changelog

Documentation will be updated together with the changes from https://github.com/PennyLaneAI/pennylane/pull/5116 in a follow-up PR

Builds on top of https://github.com/PennyLaneAI/pennylane/pull/5116
[sc-54959]
[sc-54955]

### CODE REVIEW

**Review by albi3ro — APPROVED** [review]
🎉  🚀 Thank you so much!

File: tests/ops/op_math/test_sum.py
Comment (trbromley) [inline tests/ops/op_math/test_sum.py:1]: Could we have a test that confirms the copyability? I.e., that you can execute the repr and get a valid PL op?

Comment (Qottmann) [reply tests/ops/op_math/test_sum.py:1]: Added some tests. Noticed that `qml.equal()` does not always correctly yield the right answer due to nesting (which is due to the lazy execution).

Opened an issue for the specific problem https://github.com/PennyLaneAI/pennylane/issues/5162

Comment (Qottmann) [reply tests/ops/op_math/test_sum.py:1]: Should not be a blocker for this PR though, `eval(repr(op))` and `op` _are_ the same operators, this is more a problem of pennylane utility functionality for testing (i.e. qml.equal)

**Comment by albi3ro** [comment]
Screenshot 2024-02-08 at 10 54 39 AM

**Pull request merged**